extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudParsingHelper.java

```diff
@@ -185,8 +190,20 @@ public static String resolveUrlWithEmbedPlayer(final String apiUrl) throws IOExc
      */
     public static String resolveIdWithWidgetApi(final String urlString) throws IOException,
             ParsingException {
-        // Remove the tailing slash from URLs due to issues with the SoundCloud API
         String fixedUrl = urlString;
+
+        // if URL is an on.soundcloud link, do a request to resolve the redirect
+
+        if (ON_URL_PATTERN.matcher(fixedUrl).find()) {
+            try {
+                fixedUrl = NewPipe.getDownloader().head(fixedUrl).latestUrl();
```

Comment:
The latest URL we get contains tracking parameters. From the URL you linked in the issue you opened (#1178, `https://on.soundcloud.com/a8QmYdMnmxnsSTEp9`), I get redirected to:

`https://soundcloud.com/katagiri/ultimate-dragon-forest-kingdom-64-definitive-edition-game-of-the-year-mix?ref=clipboard&p=i&c=0&si=A1822F15965A4D9B8EEBDE9DC7524F10&utm_source=clipboard&utm_medium=text&utm_campaign=social_sharing`

You should remove them by removing any query parameter (I don't think they are useful in other places like artists, playlists and sets, but please correct me if I am wrong). This is not an issue for now as we do not support "private" tracks, i.e. tracks with secret tokens.

Reply:
oh, good catch! I guess this isn't explicitly necessary since the existing tests pass but to remove tracking and such I do agree this is good.